### PR TITLE
Fixes and additions to the StaminaUsage section

### PR DESCRIPTION
**Pull request by AccretionCD**
Closed PR #160 as changes go beyond my original intentions.

Fixed the previous issues regarding stamina drain being reduced for all activities when affected weapons/tools are equipped; and added an option to adjust stamina drain for shields.
I've tested all weapon types and tools, building, repairing and terrain manipulation while jumping, sprinting, etc. so it should all be in order.

For the UseStamina patch, I've had to have it run a stack trace to determine the calling method as the comment pointed out. I'm not sure how doing it in such a way would affect performance, please let me know if there are any issues. 🙂

### CODE REVIEW

**Review by Mixone-FinallyHere — COMMENTED** [review]
Looks good to me
@radmint want to double check?

File: ValheimPlus/PlayerStats.cs [hunk 1]
```diff
@@ -29,89 +29,111 @@ private static void Postfix(Player __instance)
...
-                    case "Pickaxes":
-                        v = v - (v * (Configuration.Current.StaminaUsage.pickaxes) / 100);
-                        break;
-                    default:
-                        break;
+                    if (weaponType != "")
+                    {
+                        switch (weaponType)
+                        {
+                            case "Swords":
```
Comment (radmint) [inline ValheimPlus/PlayerStats.cs:64]: Can we use the `Skills.SkillType` enum for comparisons here for better future base game compatibility?

File: ValheimPlus/PlayerStats.cs [hunk 2]
```diff
@@ -29,89 +29,111 @@ private static void Postfix(Player __instance)
...
-                if (isHammer)
+    [HarmonyPatch(typeof(Player), "UseStamina")]
+    public static class ChangeStaminaUsageOfTools
+    {
+        private static void Prefix(ref Player __instance, ref float v)
+        {
+            if (Configuration.Current.StaminaUsage.IsEnabled)
+            {
+                string methodName = new StackTrace().GetFrame(2).GetMethod().Name;
+                if (methodName == "UpdatePlacement" || methodName == "Repair")
```
Comment (radmint) [inline ValheimPlus/PlayerStats.cs:108]: instead of these strings, could we run something like `methodName.Equals(nameof(Player.UpdatePlacement))`?

**Review by radmint — CHANGES_REQUESTED** [review]
Excellent use of calling method, thank you for describing how you tested. This looks great, just requesting small updates and I'd be happy to approve. Thank you for the improvement!

**Comment by radmint** [comment]
Oh @AccretionCD and please add your credits to the README.md

**Comment by AccretionCD** [comment]
Thank you for taking the time to review. I'll get right on the changes. 👍

File: ValheimPlus/PlayerStats.cs (hunk 1 above)
Comment (AccretionCD) [reply ValheimPlus/PlayerStats.cs:64]: Done.

File: ValheimPlus/PlayerStats.cs (hunk 2 above)
Comment (AccretionCD) [reply ValheimPlus/PlayerStats.cs:108]: Reviewing this section led me to discover an oversight with blocking which I fixed. Previously only affected blocking with a shield and not with any other weapon.
Thank you 😄

**Comment by AccretionCD** [comment]
I've made the requested changes + a small fix. Should all be in order.

**Review by radmint — DISMISSED** [review]
Looks great! Thank you!

**Comment by radmint** [comment]
@AccretionCD just needs updating from our latest push and it's good to go! Thanks again!

**Comment by AccretionCD** [comment]
Merged and tested as per usual. Should be good. 🙂

**Comment by radmint** [comment]
@AccretionCD would you mind merging master in so that your PR doesn't show 49 files changed? Thank you!

**Comment by AccretionCD** [comment]
@radmint Done. 👍

**Review by Mixone-FinallyHere — APPROVED** [review]
LGTM!

**Comment by nxPublic** [comment]
Very well done!

**Review by zedle — APPROVED** [review]

**Comment by zedle** [comment]
Looks good, merging.

**Pull request merged**